keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -40,12 +46,26 @@ class ApplicationService(
     var resources = repository.getSummaryByApplication(application)
 
     resources = resources.map { summary ->
-      if (pauser.resourceIsPaused(summary.id)) {
-        // we only update the status if the individual resource is paused,
-        // because the application pause is reflected in the response as a top level key.
-        summary.copy(status = ResourceStatus.PAUSED)
-      } else {
-        summary
+      summary.resource.let { resource ->
+        if (resource.spec is ComputeResourceSpec) {
+          resolvers.supportingComputeResources(resource as Resource<ComputeResourceSpec>)
+            .fold(resource) { computeResource, resolver ->
+              log.debug("Applying ${resolver.javaClass.simpleName} to ${computeResource.id}")
+              resolver(computeResource)
```

Comment:
This is what fills in the details about the desired artifact and version into the compute resource spec.